text/0094-remove-kv-wal.md

```diff
@@ -0,0 +1,107 @@
+# Remove KV WAL
+
+- RFC PR: https://github.com/tikv/rfcs/pull/94
+- Tracking Issue: https://github.com/tikv/repo/issues/0000
+
+## Summary
+
+Disable KV WAL completely and move all states from KV DB to raft engine.
+
+## Motivation
+
+We already have all the raft logs required for recovery, it's a waste to write everything to WAL again. Although we are using cache write almost the whole time, but if a page is marked to flush, it can still trigger random latency. And if we want to use multiple rocksdb instances in one node, writing WAL can introduce a lot of random writes.
+
+## Detailed design
+
+This RFC assumes you have know how RocksDB works and recovers from restarting.
+
+### Relation between sequence number and (region ID, apply index)
+
+In RocksDB, WAL plays the role of recording both data and order. MANIFEST plays the role of recording persistent data. Now that WAL is removed, we need a new component to replace its role. Raft logs already have all the necessary data and partial order within the same peer, and the order across peers is traced by region version. So all we need to do is record the exact index KV DB belongs to.
```

Comment:
Should elaborate on the issue why need to record this. One can simply think just replaying from the previous apply index to the commit index is enough, without realizing the apply index is stored in RAFT CF of kvdb and different CFs are flushed separately.